client/ayon_core/plugins/publish/collect_anatomy_instance_data.py

```diff
@@ -313,8 +313,9 @@ def fill_anatomy_data(self, context):
 
             # Define version
             version_number = None
-            if self.follow_workfile_version:
-                version_number = context.data("version")
+            if self.follow_workfile_version or instance.data.get(
+                "follow_workfile_version", self.follow_workfile_version):
+                    version_number = context.data("version")
```

Comment:
Just wanted to remark this logic seems very similar to what I implemented here: https://github.com/ynput/ayon-core/pull/295/files#diff-7df3c84c641d94efac636cfd13f276c9a0bea4e3a35aa36e62f74f370bee12ecR317-R323

But in that case there are certain instances that can't be matched to the workfile versions and have to version independently due to how they work - as such, it's still not artist choice but a pipeline implementation detail.